### PR TITLE
ENH Guess %Y-%m format

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Broken off from https://github.com/pandas-dev/pandas/pull/49024, opening it separately as it's small and self-contained

### CODE REVIEW

File: pandas/tests/tslibs/test_parsing.py [hunk 1]
```diff
@@ -148,6 +148,8 @@ def test_parsers_month_freq(date_str, expected):
         ("20111230", "%Y%m%d"),
         ("2011-12-30", "%Y-%m-%d"),
         ("2011", "%Y"),
+        ("2011-01", "%Y-%m"),
+        ("2011/01", "%Y/%m"),
```
Comment (WillAyd) [inline pandas/tests/tslibs/test_parsing.py:152]: So this is ISO 8601 conformant or is it just a byproduct of the implementation? Don't have the actual spec just looking at the Wikipedia. Didn't see anything for this

Comment (MarcoGorelli) [reply pandas/tests/tslibs/test_parsing.py:152]: It's not strictly ISO8601 (which wants a hyphen), but in pandas it's still treated as if it were, so long as the separator is consistent

https://github.com/pandas-dev/pandas/blob/d3adfe5a3aea0cad36b578f65d03c4f559909403/pandas/_libs/tslibs/parsing.pyx#L883-L892

https://github.com/pandas-dev/pandas/blob/009c4c622c739941ae3dfde4bc4117eb25172500/pandas/_libs/tslibs/src/datetime/np_datetime_strings.c#L87

Comment (WillAyd) [reply pandas/tests/tslibs/test_parsing.py:152]: Ah interesting...might be worth reconsidering but definitely out of scope for this PR

**Review by WillAyd — APPROVED** [review]
lgtm

**Pull request merged**